core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheManager.java

```diff
@@ -156,16 +165,22 @@ public static LocalCacheManager create(AlluxioConfiguration conf) throws IOExcep
    * @param pageStore the page store manages the cache data
    */
   @VisibleForTesting
-  LocalCacheManager(AlluxioConfiguration conf, MetaStore metaStore,
-      PageStore pageStore, CacheEvictor evictor) {
+  LocalCacheManager(AlluxioConfiguration conf, MetaStore metaStore, PageStore pageStore,
+      CacheEvictor evictor) {
     mMetaStore = metaStore;
     mPageStore = pageStore;
     mEvictor = evictor;
     mPageSize = conf.getBytes(PropertyKey.USER_CLIENT_CACHE_PAGE_SIZE);
+    mAsyncWrite = conf.getBoolean(PropertyKey.USER_CLIENT_CACHE_ASYNC_WRITE_ENABLED);
     mCacheSize = pageStore.getCacheSize();
     for (int i = 0; i < LOCK_SIZE; i++) {
       mPageLocks[i] = new ReentrantReadWriteLock();
     }
+    mPendingRequests = new ConcurrentHashSet<>();
+    mAsyncCacheExecutor = mAsyncWrite
+        ? new ThreadPoolExecutor(0, conf.getInt(PropertyKey.USER_CLIENT_CACHE_ASYNC_WRITE_THREADS),
```

Comment:
Since we are using a synchronous queue, should we set core threads to # of async write threads?

Reply:
Changed